.ci/containers/terraform-tester/test_terraform.sh

```diff
@@ -18,10 +18,43 @@ else
     exit 1
 fi
 
-scratch_path=https://$github_username:$GITHUB_TOKEN@github.com/$github_username/$gh_repo
-local_path=$GOPATH/src/github.com/terraform-providers/$gh_repo
+
+new_branch="auto-pr-$pr_number"
+old_branch="auto-pr-$pr_number-old"
+
+# https://docs.github.com/en/rest/reference/repos#compare-two-commits
+echo "Fetching diff from Github"
+http_response=$(curl \
+  -X GET \
+  -u "$github_username:$GITHUB_TOKEN" \
+  -H "Accept: application/vnd.github.v3+json" \
+  -o response.txt \
+  -w "%{http_code}" \
+  "https://api.github.com/repos/$github_username/$gh_repo/compare/$old_branch...$new_branch")
```

Comment:
Is this going to be different than the result of `git diff HEAD~1` when the PR branch is checked out?

Reply:
This is doing the same comparison as the github differ: https://github.com/GoogleCloudPlatform/magic-modules/blob/e7cdc818a67e06d0dc07d7055b3051f7f8b08825/.ci/containers/github-differ/generate_comment.sh#L42 (Three dots vs two dots doesn't seem to make a practical difference in this case - the non-API view can do either, but the API view only supports three dots.)

I think this _should_ be the same as `git diff HEAD~1` in most cases. Since both the old & new are generated, it seems like there's a bit of a risk of an old branch diverging from the master branch (resulting in a different diff) but actually for this purpose I think we should be fine to ignore that case.

We only care whether there's a new go file that will be added to the downstream, which `git diff HEAD~1` should catch, since generation always creates exactly one commit on top of the current downstream.

Reply:
TLDR I'll adjust this to check locally with `git diff HEAD~1`